lib/matplotlib/axes/_axes.py

```diff
@@ -46,9 +48,49 @@
 is_sequence_of_strings = cbook.is_sequence_of_strings
 
 
+def _plot_args_replacer(args, data):
+    if len(args) == 1:
+        return ["y"]
+    elif len(args) == 2:
+        # this can be two cases: x,y or y,c
+        if not args[1] in data:
+            # this is not in data, so just assume that it is something which
+            # will not get replaced (color spec or array like).
+            return ["y", "c"]
+        # it's data, but could be a color code like 'ro' or 'b--'
+        # -> warn the user in that case...
+        arg2 = args[1]
+        if is_string_like(arg2) and len(arg2) <= 3:
+            # all possible linestyles and color codes -> see doc of plot
+            reserved_ls = ["-", "--", "-.", ":", ".", ",", "o", "v", "^", "<",
```

Comment:
This still does not catch everything as `ro-` is valid.  I suggest calling https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_base.py#L43 , catching and ignoring `ValueErrors` and if everything comes back `None`.